### PR TITLE
[Resource] Integrated Translations

### LINKED ISSUE #3750: Merge Translation & TranslationBundle into Resource?
Hey folks!

I would like to propose merging Translation and TranslationBundle into Resource and ResourceBundle respectively. 

Caveats:

* You will still be able to creates resources which are not translatable. Just implement TranslatableInterface or not and configure ``translation`` node or not.

Rationale:

Pros:
* Removes overhead of having 4 packages instead of 2;
* Removes overhead of decoupling configuration and its processing in ResourceBundle;
* We can use TranslatableInterface and other interfaces safely everywhere to check if something supports translations, now we need to use string and check if bundle exists etc. - really annoying;
* Translation & TranslationBundle without ResourceBundle do not make much sense, is anyone actually using them without Resource?

Cons:
* I don't see any right now, but opening this RFC to find out. :)

I know that in perfect scenario we create an extensions system for ResourceBundle and every extension could add its own configuration node to every resource and also process metadata in its own way etc. and I'd love to do that but at this point we don't have time and I really don't see a reason to have it decoupled. It only adds overhead.

If someone thinks of using Translation from Sylius standalone, there are already bundles/components which have similar level of functionality.

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no|yes
| BC breaks?    | yes
| Deprecations? | no
| Related tickets | fixes #3750
| License       | MIT
| Doc PR        | -

Based on #3854

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 CHANGELOG
 =========
 
+### v0.17.0
+
+* Integrated TranslationBundle.
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/CHANGELOG.md]: Merged with TranslationBundle.

File: src/Sylius/Bundle/ResourceBundle/composer.json [hunk 2]
```diff
@@ -51,6 +51,8 @@
         "phpspec/phpspec": "^2.4",
         "phpunit/phpunit": "^4.8",
         "sensio/generator-bundle": "^3.0",
+        "symfony/form": "^2.7",
+        "symfony/validator": "^2.7",
         "sylius/rbac-bundle": "0.17.*",
         "sylius/translation-bundle": "0.17.*",
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/composer.json]: This should be removed.

File: src/Sylius/Bundle/ArchetypeBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -82,7 +82,7 @@ private function addResourcesSection(ArrayNodeDefinition $node)
                                             ->scalarNode('model')->defaultValue(Archetype::class)->cannotBeEmpty()->end()
                                             ->scalarNode('interface')->defaultValue(ArchetypeInterface::class)->cannotBeEmpty()->end()
                                             ->scalarNode('controller')->defaultValue(ResourceController::class)->cannotBeEmpty()->end()
-                                            ->scalarNode('repository')->defaultValue(TranslatableResourceRepository::class)->cannotBeEmpty()->end()
+                                            ->scalarNode('repository')->defaultValue(TranslatableRepository::class)->cannotBeEmpty()->end()
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ArchetypeBundle/DependencyInjection/Configuration.php]: Repository should not have default value at all

File: src/Sylius/Bundle/ContactBundle/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -111,7 +111,7 @@ private function addResourcesSection(ArrayNodeDefinition $node)
                                         ->scalarNode('model')->defaultValue(Topic::class)->cannotBeEmpty()->end()
                                         ->scalarNode('interface')->defaultValue(TopicInterface::class)->cannotBeEmpty()->end()
                                         ->scalarNode('controller')->defaultValue(ResourceController::class)->cannotBeEmpty()->end()
-                                        ->scalarNode('repository')->defaultValue(TranslatableResourceRepository::class)->cannotBeEmpty()->end()
+                                        ->scalarNode('repository')->defaultValue(TranslatableRepository::class)->cannotBeEmpty()->end()
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ContactBundle/DependencyInjection/Configuration.php]: Repository should not have default value at all

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -144,4 +145,18 @@ private function addSettingsSection(ArrayNodeDefinition $node)
...
     }
+
+    /**
+     * @param ArrayNodeDefinition $node
+     */
+    private function addTranslationsSection(ArrayNodeDefinition $node)
+    {
+        $node
+            ->children()
+            ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->end()
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Configuration.php]: These nodes should be indented more :)

**Comment by lchrusciel** [comment]
I think it should be based on #3854, because `TranslatableResourceRepository ` is removed totally there.

File: src/Sylius/Bundle/SequenceBundle/Doctrine/ORM/NumberListener.php [hunk 6]
```diff
@@ -28,7 +29,7 @@
  * @author Alexandre Bacco <alexandre.bacco@gmail.com>
  * @author Saša Stamenković <umpirsky@gmail.com>
  */
-class NumberListener
+class NumberListener implements NumberListenerInterface
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/SequenceBundle/Doctrine/ORM/NumberListener.php]: Listeners should not use interfaces. :)

**Comment by pjedrzejewski** [comment]
@NeverResponse Requires rebasing after merge of #3854.

**Comment by pjedrzejewski** [comment]
Thanks Jasiek! :+1:

**Pull request merged**